packages/badge/src/interfaces.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  ITheme,
+  IThemeNamespace,
+  ICSSProperties,
+} from '@xcritical/theme';
+
+import { DetailedHTMLProps, HTMLAttributes } from 'react';
+
+
+export interface IBaseBadgeTheme extends ICSSProperties {
+  inverse?: ICSSProperties;
+}
+
+export type BadgeTheme = ITheme<IBaseBadgeTheme>;
+
+export interface IThemeBadgeProps {
+  theme?: IThemeNamespace<BadgeTheme>;
+  appearance?: string;
+  baseAppearance?: string;
+  inverse?: boolean;
+}
+
+
+export interface IBadgeProps extends IThemeBadgeProps,
+  DetailedHTMLProps<HTMLAttributes<HTMLSpanElement>, HTMLSpanElement> {
```

Comment:
`export interface IBadgeProps extends IThemeBadgeProps, HTMLSpanElement {}`
I think that would be enough.

Reply:
Nope, because HTMLSpanElement contains props and DOM functions like: `appendChild`, `click` and other.

All native elements in react or styled-components extend from `JSX.IntrinsicElements` (see [docs](https://www.typescriptlang.org/docs/handbook/jsx.html#intrinsic-elements))
But, `JSX.IntrinsicElements.span` inside looks like what I wrote. Ok, I will can use `JSX.IntrinsicElements`. I think it will be better